rlclientlib/logger/logger_facade.cc

```diff
@@ -64,39 +103,67 @@ namespace reinforcement_learning {
       const std::vector<std::vector<float>>& pdfs, const std::string& model_version, api_status* status) {
       switch (version) {
         case 1: return v1->log_decision(event_id, context, flags, action_ids, pdfs, model_version, status);
+        case 2: return v2->log(event_id.c_str(), serializer.event(context, flags, action_ids, pdfs, model_version), serializer.type, status);
         default: return protocol_not_supported(status);
       }
     }
 
     observation_logger_facade::observation_logger_facade(const utility::configuration& c, i_message_sender* sender, utility::watchdog& watchdog, i_time_provider* time_provider, error_callback_fn* perror_cb)
     : version(c.get_int(name::PROTOCOL_VERSION, value::DEFAULT_PROTOCOL_VERSION))
-    , v1(version == 1 ? new observation_logger(c, sender, watchdog, time_provider, perror_cb) : nullptr) {
+    , v1(version == 1 ? new observation_logger(c, sender, watchdog, time_provider, perror_cb) : nullptr)
+    , v2(version == 2 ? new generic_event_logger(
+      sender,
+      c.get_int(name::OBSERVATION_SEND_HIGH_WATER_MARK, 198 * 1024),
+      c.get_int(name::OBSERVATION_SEND_BATCH_INTERVAL_MS, 1000),
+      c.get_int(name::OBSERVATION_SEND_QUEUE_MAX_CAPACITY_KB, 16 * 1024) * 1024,
+      c.get(name::QUEUE_MODE, "DROP"),
+      watchdog,
+      time_provider,
+      perror_cb) : nullptr) {
     }
 
     int observation_logger_facade::init(api_status* status) {
       switch (version) {
         case 1: return v1->init(status);
+        case 2: return v2->init(status);
         default: return protocol_not_supported(status);
       }
     }
 
     int observation_logger_facade::log(const char* event_id, float outcome, api_status* status) {
       switch (version) {
         case 1: return v1->log(event_id, outcome, status);
+        case 2: return v2->log(event_id, serializer.event(outcome), serializer.type, status);
         default: return protocol_not_supported(status);
       }
     }
 
     int observation_logger_facade::log(const char* event_id, const char* outcome, api_status* status) {
       switch (version) {
         case 1: return v1->log(event_id, outcome, status);
+        case 2: return v2->log(event_id, serializer.event(outcome), serializer.type, status);
+        default: return protocol_not_supported(status);
+      }
+    }
+
+    int observation_logger_facade::log(const char* event_id, int index, float outcome, api_status* status) {
```

Comment:
Same question, only v2 here, why? Where is this exposed?

Reply:
Please see the answer above. Not exposed yet.